RecoTracker/MkFitCMS/interface/LayerNumberConverter.h

```diff
@@ -1,21 +1,39 @@
 #ifndef RecoTracker_MkFitCMS_interface_LayerNumberConverter_h
 #define RecoTracker_MkFitCMS_interface_LayerNumberConverter_h
 
+#include <exception>
+
 namespace mkfit {
 
-  enum struct TkLayout { phase0 = 0, phase1 = 1 };
+  enum struct TkLayout { phase0 = 0, phase1 = 1, phase2 = 2 };
 
   class LayerNumberConverter {
   public:
     LayerNumberConverter(TkLayout layout) : lo_(layout) {}
+    void reset(TkLayout layout) { lo_ = layout; }
     unsigned int nLayers() const {
       if (lo_ == TkLayout::phase0)
         return 69;
       if (lo_ == TkLayout::phase1)
         return 72;
+      if (lo_ == TkLayout::phase2)
+        return 60;  // 4 + 12 + 2*(12 + 10) = 16 + 22 + 22 = 60
       return 10;
     }
+    TkLayout getEra() const { return lo_; }
     int convertLayerNumber(int det, int lay, bool useMatched, int isStereo, bool posZ) const {
+      if (lo_ == TkLayout::phase2) {
+        if (det == 1)
+          return lay - 1;
+        if (det == 2)
+          return 16 + lay - 1 + (posZ ? 0 : 22);
+        if (det == 5)
+          return 4 + (2 * (lay - 1)) + isStereo;
+        if (det == 4)
+          return 16 + 12 + (2 * (lay - 1)) + isStereo + (posZ ? 0 : 22);
+        throw std::runtime_error("bad subDet");
```

Comment:
`cms::Exception` is usually preferred. Is there a way to insert it with ifdefs?
https://github.com/cms-sw/cmssw/blob/master/FWCore/Utilities/interface/Exception.h

Reply:
There are many other instances of std::exceptions being thrown in MkFitXyz files that are not included in this PR. Also, semantics of std::exceptions is different than that of cms::Exception, i.e., `std::exception_type("message")` vs. `cms::Exception("exception_type", "message")` so it is not a simple question of defining/using the preferred exception type.

As fixing / addressing it here will go out of scope of this PR, can we open a new issue and follow up in detail there?

Reply:
> As fixing / addressing it here will go out of scope of this PR, can we open a new issue and follow up in detail there?

OK with me

Reply:
Issue created: #39983